driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -216,25 +217,26 @@ public Statement saveQuery(T entity, Option... options) {
     }
 
     private ListenableFuture<BoundStatement> saveQueryAsync(T entity, final EnumMap<Option.Type, Option> options) {
-        final Map<AliasedMappedProperty, Object> values = new HashMap<AliasedMappedProperty, Object>();
-        boolean saveNullFields = shouldSaveNullFields(options);
+        final Map<AliasedMappedProperty, Object> columnToValue = new HashMap<AliasedMappedProperty, Object>();
+        final boolean useUnsetForNullValue = !shouldSaveNullFields(options) && manager.protocolVersionAsInt >= 4;
+        final boolean includeColumnsWithNullValue = shouldSaveNullFields(options) || useUnsetForNullValue;
```

Comment:
> It's not possible to use the modifiesQueryString method to without some serious rewriting/refactor. The reason being that when generating the prepared statement in QueryType.makePreparedQueryString you need the values for the columns so you can omit the columns with null values when generating the statement to prepare to support <ProtocolV4.

It seems like `modifiesQueryString` wasn't relevant anyways.  This is where the magic is happening, very clever!     👍